src/index/updater/inscription_updater.rs

```diff
@@ -178,7 +178,7 @@ impl<'a, 'db, 'tx> InscriptionUpdater<'a, 'db, 'tx> {
       Ok(self.reward - output_value)
     } else {
       self.flotsam.extend(inscriptions.map(|flotsam| Flotsam {
-        offset: self.reward + flotsam.offset,
+        offset: self.reward + flotsam.offset - output_value,
```

Comment:
This is a bug fix, so it needs a test to ensure that we don't change it and regress. The test should fail without this change, and pass with this change. Please see other tests for an idea for how to do this.

Reply:
I modified the existing test for inscriptions spent as fees. Bug #1841 only happens when there are no outputs in the tx that spends the inscription as fee, so adding any input triggers the bug.